### PR TITLE
[extractor/beatbump] Add extractor

### LINKED ISSUE #4653: Please add support for beatbump.ml
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a new site support request
- [X] I've verified that I'm running yt-dlp version **2022.08.08** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that none of provided URLs [violate any copyrights](https://github.com/ytdl-org/youtube-dl#can-you-add-support-for-this-anime-video-site-or-site-which-shows-current-movies-for-free) or contain any [DRM](https://en.wikipedia.org/wiki/Digital_rights_management) to the best of my knowledge
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and am willing to share it if required

### Region

Wordwide (I suppose)

### Example URLs

Single audio: `https://beatbump.ml/listen?id=lYBUbBu4W08`
Playlist: `https://beatbump.ml/playlist/VLPL2MI040U_GXq1L5JUxNOulWCyXn-7QyZK`
Artist: `https://beatbump.ml/artist/UCwZEU0wAwIyZb4x5G_KJp2w`

### Provide a description that is worded well enough to be understood

Beatbump.ml is an opensource alternative to youtube music (music.youtube.com), in the same way that invidious (instances) is an opensource alternative to youtube. And since yt-dlp already supports invidious, I think it would be nice to add support for beatbump.ml as well

Afaik, beatbump.ml is the only instance of it right now and its source code is located here
https://github.com/snuffyDev/Beatbump

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
$ yt-dlp -vU https://beatbump.ml/listen?id=lYBUbBu4W08
[debug] Command-line config: ['-vU', 'https://beatbump.ml/listen?id=lYBUbBu4W08']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.08.08 [3157158f7]
[debug] Python 3.10.5 (CPython 64bit) - Linux-5.18.0-3-amd64-x86_64-with-glibc2.33 (glibc 2.33)
[debug] Checking exe version: ffmpeg -bsfs
[debug] Checking exe version: ffprobe -bsfs
[debug] exe versions: ffmpeg 5.1 (fdk,setts), ffprobe 5.1
[debug] Optional libraries: Cryptodome-3.11.0, brotli-1.0.9, certifi-2020.06.20, mutagen-1.45.1, sqlite3-2.6.0, websockets-10.2
[debug] Proxy map: {}
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2022.08.08, Current version: 2022.08.08
yt-dlp is up to date (2022.08.08)
[debug] [generic] Extracting URL: https://beatbump.ml/listen?id=lYBUbBu4W08
[generic] listen?id=lYBUbBu4W08: Downloading webpage
WARNING: [generic] Falling back on generic information extractor
[generic] listen?id=lYBUbBu4W08: Extracting information
[debug] Looking for Brightcove embeds
[debug] Looking for embeds
ERROR: Unsupported URL: https://beatbump.ml/listen?id=lYBUbBu4W08
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/yt_dlp/YoutubeDL.py", line 1441, in wrapper
    return func(self, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/yt_dlp/YoutubeDL.py", line 1517, in __extract_info
    ie_result = ie.extract(url)
  File "/usr/lib/python3/dist-packages/yt_dlp/extractor/common.py", line 666, in extract
    ie_result = self._real_extract(url)
  File "/usr/lib/python3/dist-packages/yt_dlp/extractor/generic.py", line 3077, in _real_extract
    raise UnsupportedError(url)
yt_dlp.utils.UnsupportedError: Unsupported URL: https://beatbump.ml/listen?id=lYBUbBu4W08
```

### PR DESCRIPTION
**Pull request by Bobscorn**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

</details>

Adds an Information Extractor for beatbump.ml videos, artists, playlists and albums.

Fixes #4653

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Fix or improvement to an extractor (Make sure to add/update tests)
- [x] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

### CODE REVIEW

File: yt_dlp/extractor/beatbump.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
+import re
+
+from yt_dlp.utils import ExtractorError
```
Comment (Lesmiscore) [inline yt_dlp/extractor/beatbump.py:3]:
```suggestion
from ..utils import ExtractorError, parse_qs
```

File: yt_dlp/extractor/beatbump.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+from .common import InfoExtractor
+
+
+class BeatBumpIE(InfoExtractor):
```
Comment (Lesmiscore) [inline yt_dlp/extractor/beatbump.py:7]: Split this into 4 extractors, for each branch

File: yt_dlp/extractor/beatbump.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+            'creator': 'Eagles',
+            'description': 'md5:1846212e0e373e0970f844516a9ced39',
+            'playable_in_embed': True,
+            'uploader_url': 'http://www.youtube.com/channel/UC49r4GNHHpc-eQ9hmD2Rg6A',
+            'album': 'Legacy',
+            'age_limit': 0,
+            'channel': 'Eagles',
+            'availability': 'public',
+        }
+    }, {
```
Comment (Lesmiscore) [inline yt_dlp/extractor/beatbump.py:42]: 1. Add `'add_ie': ['Youtube'],` for this test and `'add_ie': ['YoutubeTab'],` for others
2. Remove the whole `info_dict` for all tests; they're not from this IE

Comment (coletdjnz) [reply yt_dlp/extractor/beatbump.py:42]: > Remove the whole info_dict for all tests; they're not from this IE

I think this is fine - it needs to test that the url is sent to the youtube extractor correctly and that it understands it. Tests that resolve to other extractors are fairly common.

File: yt_dlp/extractor/beatbump.py (hunk 2 above)
Comment (Bobscorn) [reply yt_dlp/extractor/beatbump.py]: Done in 13c7367e5f54cbbf91cac022e39c7933d37198c2

**Pull request merged**